helpers/subscriptionControllers.js

```diff
@@ -0,0 +1,140 @@
+import { cssEscape } from '../../helpers/dom.js';
+
+export class ProviderController {
```

Comment:
For later: I don't love this name since it overlaps with our `RequesterMixin`/`ProviderMixin` and could cause some confusion.

Reply:
Yeah I tried out lots of different names and didn't like most of them.  We could do publisher/subscriber, but that's an actual model that this doesn't seem to _really_ follow.  Sender/Receiver maybe, but that seems more like a 1-to-1 relationship.  Sender/Subscriber?

Reply:
I don't mind "provider" but maybe just add more context to it? Similar to `ForAttributeSubscriberController` -- it's a controller for subscribing to the "for" attribute of a host.

Reply:
`SubscriptionProviderController`, `EventSubscriberController`, and `IdSubscriberController`?

Reply:
I think those work, ya! Also wondering if we should start a `controllers` top-level directory... they're similar to mixins in that sense.